src/patroni.py

```diff
@@ -306,13 +306,30 @@ def member_started(self) -> bool:
             allow server time to start up.
         """
         try:
-            for attempt in Retrying(stop=stop_after_delay(90), wait=wait_fixed(3)):
+            for attempt in Retrying(stop=stop_after_delay(30), wait=wait_fixed(3)):
                 with attempt:
                     r = requests.get(f"{self._patroni_url}/health", verify=self._verify)
         except RetryError:
-            return False
+            logger.debug(f"Patroni health check failed! Was TLS enabled? {self._tls_enabled}")
+            # Try reaching the endpoint with TLS inverted
+            try:
+                self._tls_enabled = not self._tls_enabled
+                for attempt in Retrying(stop=stop_after_delay(30), wait=wait_fixed(3)):
+                    with attempt:
+                        r = requests.get(f"{self._patroni_url}/health", verify=self._verify)
+            except RetryError:
+                logger.debug(
+                    f"Patroni health check failed still, even with TLS inverted: {self._tls_enabled}"
+                )
+                self._tls_enabled = not self._tls_enabled
+                return False
 
-        return r.json()["state"] in RUNNING_STATES
+        if r.json()["state"] in RUNNING_STATES:
+            logger.debug("Patroni health check succeeded.")
+            return True
+        else:
+            logger.debug(f'Patroni online, but unit in bad state: {r.json()["state"]}')
+            return False
```

Comment:
I have noticed, the patroni often replies 500 the very first request, and then 200 all the following requests the next second... and we reject/defer the whole event processing due to those one 500. Maybe we can save out time here to accept the very first 500 once OR at least try to improve this upstream and have no 500 from patroni on boot.

Reply:
We normally retry the call at least 10 times here before giving up, no? When you say the first request, you mean this first batch of requests+retries? For me it seems a little risky to accept 500 as a valid response, probably worth trying to understand this better. Have any examples on CI/ticket for it? I'll keep an eye

Reply:
Here is the pricese example of my story in 14/edge: https://pastebin.canonical.com/p/HXGpNqnsNJ/

As you can see we reloaded patroni and hit 503 in line 25 => deffered => timeout for deployment until the juju res-send event. Unnecessary waiting.... :sob:

Reply:
I created the following ticket with the information we have on this at the moment: https://warthogs.atlassian.net/browse/DPE-4728
We should probably consider some alternatives, I may have some ideas on this, will keep an eye!